components/tabs/README.md

```diff
@@ -37,77 +44,217 @@ Tabs are used to present related information in mutually exclusive panels, allow
+
+### Removing a Tab
+
+A `tab` can be removed from the DOM using any regular method (e.g., `removeChild`). In order for the removal animation to be shown, the `hideTab` helper method in `d2l-tabs` must be used. An example of this is shown below, which waits for the promise in `hideTab` to resolve and then removes the tab and panel from the DOM. Ensure that when a `tab` is removed, the corresponding panel is also removed.
+
+```
+const tabs = this.shadowRoot.querySelector('d2l-tabs');
+const tab = tabs.querySelector('d2l-tab');
+Promise.resolve(tabs.hideTab((tab))).then(() => {
+  const panel = tabs.querySelector(`d2l-tab-panel[labelled-by="${tab.id}"]`);
+  if (panel) tabs.removeChild(panel);
+  tabs.removeChild(tab);
+});
+```
+
+### Custom Tabs
+
+The `TabMixin` can be used to create custom tabs. It is IMPORTANT to call the `dispatchContentChangeEvent` function in `TabMixin` when content changes in the consumer in order to properly trigger calculations.
+
+<!-- docs: demo code sandboxTitle:'TabMixin' display:block -->
+```html
+<script type="module">
+  import { css, html, LitElement, unsafeCSS } from 'lit';
+  import { getFocusPseudoClass } from '@brightspace-ui/core/helpers/focus.js';
+  import { TabMixin } from '@brightspace-ui/core/components/tabs/tab-mixin.js';
+
+  class TabCustom extends TabMixin(LitElement) {
+
+    static get styles() {
+      const styles = [ css`
+        .d2l-tab-custom-content {
+          margin: 0.5rem;
+          overflow: hidden;
+          padding: 0.1rem;
+          white-space: nowrap;
+        }
+        :host(:first-child) .d2l-tab-custom-content {
+          margin-inline-start: 0;
+        }
+        :host(:${unsafeCSS(getFocusPseudoClass())}) > .d2l-tab-custom-content {
+          border-radius: 0.3rem;
+          box-shadow: 0 0 0 2px var(--d2l-color-celestine);
+          color: var(--d2l-color-celestine);
+        }
+      `];
+
+      super.styles && styles.unshift(super.styles);
+      return styles;
+    }
+
+    renderContent() {
+      return html`
+        <div class="d2l-tab-custom-content">
+          <slot @slotchange="${this.#handleSlotchange}"></slot>
+        </div>
+      `;
+    }
+
+    #handleSlotchange() {
+      this.dispatchContentChangeEvent();
+    }
+  }
+
+  customElements.define('d2l-tab-custom', TabCustom);
+</script>
+<script type="module">
+  import '@brightspace-ui/core/components/tabs/tabs.js';
+  import '@brightspace-ui/core/components/tabs/tab-panel.js';
+  import '@brightspace-ui/core/components/count-badge/count-badge.js';
+</script>
+
+<d2l-tabs text="Courses">
+  <d2l-tab-custom id="all" slot="tabs"><div style="color: purple;">All</div></d2l-tab-custom>
+  <d2l-tab-panel labelled-by="all" slot="panels">Tab content for All</d2l-tab-panel>
+  <d2l-tab-custom id="biology" slot="tabs" selected>
+    Biology <d2l-count-badge has-tooltip number="100" size="small" tab-stop text="100 new notifications" type="notification"></d2l-count-badge>
```

Comment:
Do we want to use `tab-stop` for the count badge? `d2l-tabs` has a bunch of code to manage the focus, where users are expected to navigate the list using arrow keys - does this interfere with its focus management? Like menu-items that use arrow keys, there may be an expectation (both code-wise and on the part of the user) that each item has a single focusable element... the tab itself. If there is a requirement for this, we should probably add it above in the guidance.

Reply:
Agreed - this gets the user into a confusing state. I think recommending that each tab contain only one focusable element makes sense.